hazelcast/src/test/java/com/hazelcast/client/protocol/ClientExceptionFactoryTest.java

```diff
@@ -0,0 +1,210 @@
+                new Object[]{new QuorumException(randomString())},
+                new Object[]{new ReachedMaxSizeException(randomString())},
+                new Object[]{new RejectedExecutionException(randomString())},
+                new Object[]{new RemoteMapReduceException(randomMapName(), Collections.<Exception>emptyList())},
+                new Object[]{new ResponseAlreadySentException(randomString())},
+                new Object[]{new RetryableHazelcastException(randomString())},
+                new Object[]{new RetryableIOException(randomString())},
+                new Object[]{new RuntimeException(randomString())},
+                new Object[]{new SecurityException(randomString())},
+                new Object[]{new SocketException(randomString())},
+                new Object[]{new StaleSequenceException(randomString(), 1)},
+                new Object[]{new TargetDisconnectedException(randomString())},
+                new Object[]{new TargetNotMemberException(randomString())},
+                new Object[]{new TimeoutException(randomString())},
+                new Object[]{new TopicOverloadException(randomString())},
+                new Object[]{new TopologyChangedException(randomString())},
+                new Object[]{new TransactionException(randomString())},
+                new Object[]{new TransactionNotActiveException(randomString())},
+                new Object[]{new TransactionTimedOutException(randomString())},
+                new Object[]{new URISyntaxException(randomString(), randomString())},
+                new Object[]{new UTFDataFormatException(randomString())},
+                new Object[]{new UnsupportedOperationException(randomString())},
+                new Object[]{new WrongTargetException(randomString())},
+                new Object[]{new XAException(randomString())},
+                new Object[]{new AccessControlException(randomString())},
+                new Object[]{new LoginException(randomString())},
+                new Object[]{new UnsupportedCallbackException(new Callback() {
+                })},
+                new Object[]{new NoDataMemberInClusterException(randomString())},
+                new Object[]{new ReplicatedMapCantBeCreatedOnLiteMemberException(randomString())},
+                new Object[]{new MaxMessageSizeExceeded()},
+                new Object[]{new WANReplicationQueueFullException(randomString())});
+
+    }
+
+    @Test
+    public void testException() {
+        ClientMessage exceptionMessage = exceptionFactory.createExceptionMessage(throwable);
+        ClientMessage responseMessage = ClientMessage.createForDecode(exceptionMessage.buffer(), 0);
+        Throwable resurrectedThrowable = exceptionFactory.createException(responseMessage);
+        assertEquals(throwable.getClass(), resurrectedThrowable.getClass());
+        assertArray(throwable.getStackTrace(), resurrectedThrowable.getStackTrace());
```

Comment:
Thanks. I did not know the exact naming, and my IDE did not suggest when I am trying to find out. Changing it now.